### PR TITLE
Update ml training code for split line start:end

### PR DESCRIPTION
**Pull request by babenek**
## Description

Please include a summary of the change and which is fixed.

- Remove augmentation (will be migrated to CredData) - fixes security warning #2
- Update code for ML train to use splitted rows
- Fix warnings for overlapping exceptions

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] UnitTest
- [x] Benchmark

### CODE REVIEW

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -22,7 +22,8 @@ jobs:
       - name: Checkout CredData
         uses: actions/checkout@v4
         with:
-          repository: Samsung/CredData
+          repository: babenek/CredData
```
Comment (babenek) [inline .github/workflows/benchmark.yml:25]: TODO: rollback after https://github.com/Samsung/CredData/pull/121 merge

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -21,7 +21,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ ubuntu-latest, windows-latest, macos-latest ]
+        os: [ ubuntu-latest, windows-latest, macos-13 ]
```
Comment (babenek) [inline .github/workflows/test.yml:24]:
```suggestion
        os: [ ubuntu-latest, windows-latest, macos-latest ]
```

Comment (babenek) [reply .github/workflows/test.yml:24]: will be fixed in #544

**Review by Yullia — APPROVED** [review]
Great job!

**Pull request merged**